metal/resource_metal_project_api_key.go

```diff
@@ -0,0 +1,105 @@
+package metal
+
+import (
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/packethost/packngo"
+)
+
+func metalProjectAPIKey() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"project_id": {
+			Type:     schema.TypeString,
+			ForceNew: true,
+			Required: true,
```

Comment:
This resource could be `metal_api_key`, with optional project scoping, if this field were not required.

Reply:
FWIW, I recognize that we have both `metal_project_ssh_key` and `metal_ssh_key`. 

I'm not sure that I see a reason for the distinction, then or now. `project_ssh_keys` came a few years after `ssh_keys`.
https://github.com/equinix/terraform-provider-metal/blob/main/metal/resource_metal_project_ssh_key.go

When I was first introduced to the API, I found some justification for this because some resources are created behind /projects/id/foo, while others are not.  After a bit more exposure to the API, I find that resources (project or not) are all typically accessible behind /foo/id.  Project scoped or not, these optionally scoped resources have the same response body, other than the project id field.

When a resource is project scoped, the API path contains the (optional) project_id rather than the POST body. If a listing endpoint is present in /projects/id/foo, it is offered as a convenience to /foo?project_id=id.

They are otherwise the same.

Reply:
For what it's worth, I did consider using a single resource for both and I do prefer that API, but unfortunately the backing API at EM doesn't provide a single read endpoint. I did consider working around that, but I importing a resource wouldn't work and I thought that was more important. Perhaps we can work with engineering to clean up this API. ￼

Reply:
@rawkode you're right that the read endpoint for user API key is different. You could create a resource for the API key, reusing some of the project API key code. If you'd change the project_id to `Optional: true`, you could just add `resource_metal_user_api_key.go` as

```go
func resourceMetalUserAPIKey() *schema.Resource {
	return &schema.Resource{
		Create: resourceMetalProjectAPIKeyCreate,
		Read:   resourceMetalUserAPIKeyRead,
		Delete: resourceMetalProjectAPIKeyDelete,
		Importer: &schema.ResourceImporter{
			State: schema.ImportStatePassthrough,
		},

		Schema: metalProjectAPIKey(),
	}
}

func resourceMetalUserAPIKeyRead(d *schema.ResourceData, meta interface{}) error {
	client := meta.(*packngo.Client)
	apiKey, err := client.APIKeys.UserGet(d.Id(), nil)
	if err != nil {
		err = friendlyError(err)
		if isNotFound(err) {
			log.Printf("[WARN] User APIKey (%s) not found, removing from state", d.Id())
			d.SetId("")
			return nil
		}
		return err
	}
	d.SetId(apiKey.ID)
	d.Set("description", apiKey.Description)
	d.Set("read_only", apiKey.ReadOnly)
	d.Set("created", apiKey.Created)
	d.Set("updated", apiKey.Updated)
	return nil
}
```

.. the user API key resource would actually be importable.

If you don't want to do it in this PR, it's OK we can add `metal_user_api_key` later. Thsis is good to go after you remove the Import and the Update method.